parsers/token.js

```diff
@@ -1,23 +1,13 @@
 const Overlay = require("../overlay.js");
 const ColourParser = require("./colour-parser.js");
-
-const sizeLookups = new Map([
-  ["T", "tiny"],
-  ["S", "small"],
-  ["M", "medium"],
-  ["L", "large"],
-  ["H", "huge"],
-  ["G", "gargantuan"],
-]);
-
-const sizes = ["T", "S", "M", "L", "H", "G"];
+const SizeParser = require("./size-parser.js");
 
 module.exports = class TokenParser {
-  parse(str) {
+  async parse(str) {
     if (str.length < 2) return false;
 
     // a string matching a token definition eg. D3rp-asdsa
-    const reg = /^([A-Z]{1,2}[0-9]{1,2})([TSMLHG]*?)(PK|PU|BK|GY|BN|[WKEARGBYPCNOI]|~[0-9A-f]{6}|~[0-9A-f]{3})?(-(.+))?$/i;
+    const reg = /^([A-Z]{1,2}[0-9]{1,2})([TSMLHG]*?)(PK|PU|BK|GY|BN|[WKEARGBYPCNOI]|~[0-9A-f]{6}|~[0-9A-f]{3})?(-([^-]+))?(-([^-]+))?$/i;
```

Comment:
a1-label-img 
We lose support of names with hyphens?
Is there a URL safe character that could be used instead of - ?
What happens if we want more advanced labels in the future?

Reply:
we could use ~ ?

Reply:
actually the only issue with dash is not supporting - in names
which avrae probably doesn't do anyway?
if we add a sub labels down the line I reckon do it with -- (double dash)

Reply:
~ is ok too, if you're using these tokens you're probably not typing it yourself anyway